### ISSUE TITLE
issue with quoted comment marker

### ISSUE THREAD

**Issue by GitMensch** [maelvls/vscode_autotools#14]
Some entries actual need an empty "on error do nothing" to override the default behavior (likely of stopping with an error).

A good sample is likely:

```m4
PKG_CHECK_MODULES([CJSON], [libcjson], [], [#])
```

this extension does not stop the inline comment and therefore assumes the macro parameters are not correctly closed, which makes editing all following lines bad (full of red errors and no option to match brackets correct)

**Comment by GitMensch** [comment]
Just wanted to note that this is still bugging me, I've tried to have a "more or less quick" look at the definition but neither "comment" nor "#" are good search criterias in the tmLanguage definition for autoconf. Any hint where to look at?
 I _think_ `]` will always end a line `#` comment, but I have no clue where to add that rule.

**Comment by maelvls** [comment]
So sorry I didn't get to come back to you before :(

Could you try to open a PR with the change? I would greatly appreciate it!

**Comment by GitMensch** [comment]
I'd love to pull anything but I have no clue where to do the change. Help/hints wanted... (I could then investigate and test and PR)

**Comment by maelvls** [comment]
Hmmm I think I know where the bug comes from... it does not seem to be due to the grammar itself ([Autoconf.YAML-tmLanguage](https://github.com/maelvls/vscode_autotools/blob/master/Autoconf.YAML-tmLanguage)). I would guess it comes from how VSCode parses comments:

https://github.com/maelvls/vscode_autotools/blob/dc6849fae637ebcd6b823bee4a56d9e677f9fa59/language-configuration.json#L2-L5

**Comment by maelvls** [comment]
Nope, doesn't look like an issue with `language-configuration.json`...

Works:

```m4
PKG_CHECK_MODULES([CJSON], [#])
```

Breaks:

```m4
PKG_CHECK_MODULES([CJSON], [], [#])
```

I'm clueless 🙁

**Comment by GitMensch** [comment]
Is this an upstream issue?
